s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -885,6 +884,111 @@ trait S3 { self =>
+    *                      Bucket owners need not specify this parameter in their requests.
+    * @return an [[Observable]] that emits the [[S3Object]]s.
+    */
+
+  def listLatestObject(
+    bucket: String,
+    prefix: Option[String] = None,
+    requestPayer: Option[RequestPayer] = None): Observable[S3Object] = {
+    val sort: (S3Object, S3Object) => S3Object = (a, b) =>
+      if (a.lastModified().compareTo(b.lastModified()) >= 0) a else b
+    for {
+      listResponse <- ListObjectsObservable(bucket, prefix, None, requestPayer, this.s3Client)
+      s3Object     <- Observable.fromIterable(listResponse.contents.asScala).reduce(sort)
+    } yield s3Object
+  }
+
+  /** Returns oldest N objects in bucket.
+    *
+    * ==Example==
+    *
+    * {{{
+    * import cats.effect.Resource
+    * import monix.connect.s3.S3
+    * import monix.eval.Task
+    * import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+    * import software.amazon.awssdk.regions.Region
+    * import monix.execution.Scheduler.Implicits.global
+    *
+    * val bucket = "my-bucket"
+    * val prefix = "prefix/to/list/keys"
+    * val s3AccessKey: String = "TESTKEY"
+    * val s3SecretKey: String = "TESTSECRET"
+    * val basicAWSCredentials = AwsBasicCredentials.create(s3AccessKey, s3SecretKey)
+    * val staticCredProvider = StaticCredentialsProvider.create(basicAWSCredentials)
+    *
+    * val s3: Resource[Task, S3] = S3.create(staticCredProvider, Region.AWS_GLOBAL)
+    *
+    * s3.use{s3 => s3.listOldestNObjects(bucket, 6, prefix = Some(prefix)).toListL}.runToFuture
+    * }}}
+    *
+    * To use this operation in an AWS (IAM) policy, you must have permissions to perform
+    * the `ListBucket` action. The bucket owner has this permission by default and can grant it.
+    *
+    * @param bucket        target S3 bucket name of the object to be downloaded.
+    * @param amount        the number of objects you would like to return
+    * @param prefix        limits the response to keys that begin with the specified prefix.
+    * @param requestPayer  confirms that the requester knows that she or he will be charged for
+    *                      the list objects request in V2 style.
+    *                      Bucket owners need not specify this parameter in their requests.
+    * @return an [[Observable]] that emits the [[S3Object]]s.
+    */
+
+  def listOldestNObjects(
+    bucket: String,
+    n: Int,
+    prefix: Option[String] = None,
+    requestPayer: Option[RequestPayer] = None): Observable[S3Object] = {
+    val sorted: (S3Object, S3Object) => Boolean = (x, y) =>
+      if (x.lastModified().compareTo(y.lastModified()) < 0) true else false
+    for {
+      listResponse <- ListObjectsObservable(bucket, prefix, None, None, this.s3Client)
+      s3Object <- Observable.fromIterable(listResponse.contents.asScala.foldLeft(List.empty[S3Object])((prev, curr) => {
```

Comment:
probably easier to directly fold left the `Observable`? 
It seems like as it is, it is not getting the oldest objects, but rather the oldests from the list returned on each batch request

Reply:
I'll review this and double check.

Reply:
Is this cool?: 
```
for{
    listResponse <- ListObjectsObservable(bucket,prefix,None,None, this.s3Client)
    s3ObjectList <- Observable.fromIterable(listResponse.contents.asScala).foldLeft(List.empty[S3Object])((prev, curr) => {
        prev.size match {
            case x if (x < n) => (prev :+ curr).sortWith(sorted)
            case _ =>
                prev.map(y => curr.lastModified().compareTo(y.lastModified()) < 0).contains(true) match {
                    case true => (prev.init :+ curr).sortWith(sorted)
                    case _ => prev
                }
        }
     })
    s3Objects <- Observable.fromIterable(s3ObjectList)
} yield s3Objects
```
This code at least works, but if it isn't what you want then I might be a little confused. For instance, I tried to ```Observable.fromIterable(listResponse.contents.asScala).foldLeft(Observable.empty[S3Object]```, but I don't know if there's a way to compare anything there. So if this isn't what you expected I think I'm lost.

Reply:
Sorry, I think it still not hehe
It is currently doing the fold left individually over the different emitted lists of s3Objects, and what we need is a global fold left that applies to the observable (all the s3 objects).
Something similar to this I think would work:
```
  ListObjectsObservable(bucket,prefix,None,None, s3AsyncClient).foldLeft(List.empty[S3Object])((prev, curr) => {
      (prev ++ curr.contents.asScala).sortWith(sorted).take(n)
      })
    } 
```

Reply:
Yeah, I ended up doing it from the function:
```
for{
 listResponse <- listObjects(bucket,prefix, None, None).foldLeft(List.empty[S3Object])(...)
 s3Objects <- Observable.fromIterable(listResponse)
```
However, with this latest push I ended up doing: 
```
   for {
      listResponse <- ListObjectsObservable(bucket, prefix, None, None, this.s3Client).foldLeft(List.empty[S3Object])(...)
      s3Object <- Observable.fromIterable(listResponse)
    } yield s3Object
```

I doubled checked and they both end up giving the same output but I think your example requires one less ```Observable.fromIterable``` so I ended up doing something with more your example. I also have IT:test that run and pass with a return of 1010 and it all passes.

Reply:
cool, thanks for addressing it, looks good! :)